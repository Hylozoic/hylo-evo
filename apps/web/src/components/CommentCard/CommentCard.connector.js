import { connect } from 'react-redux'
import { push } from "redux-first-history";
import { postUrl } from 'util/navigation'

export function mapStateToProps (state, props) {
  return { }
}

export function mapDispatchToProps (dispatch, props) {
  return {
    showDetails: () =>
      dispatch(push(postUrl(props.comment.post.id, props.routeParams)))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)
