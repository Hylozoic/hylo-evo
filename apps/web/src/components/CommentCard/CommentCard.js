import React from 'react'
import { useTranslation } from 'react-i18next'
import cx from 'classnames'
import RoundImage from 'components/RoundImage'
import { TextHelpers } from 'hylo-shared'
import Highlight from 'components/Highlight'
import HyloHTML from 'components/HyloHTML'
import ClickCatcher from 'components/ClickCatcher'
import CardImageAttachments from 'components/CardImageAttachments'
import CardFileAttachments from 'components/CardFileAttachments'
import classes from './CommentCard.module.scss'

export default function CommentCard ({
  comment,
  showDetails,
  expanded = true,
  highlightProps
}) {
  const { creator, post, slug, createdAt, editedAt, attachments } = comment
  const timestamp = (editedAt ? 'Edited ' : 'Commented ') + TextHelpers.humanDate(editedAt || createdAt)
  const postTitle = post.title ? TextHelpers.truncateText(post.title, 25) : TextHelpers.truncateHTML(post.details, 25)

  const commentText = expanded ? comment.text : TextHelpers.truncateHTML(comment.text, 144)
  const { t } = useTranslation()

  return (
    <span onClick={() => showDetails(comment.post.id)} className={classes.link}>
      <div className={cx(classes.commentCard, { [classes.expanded]: expanded })}>
        <div className={classes.commentHeader}>
          <RoundImage url={creator.avatarUrl} className={classes.profileImage} />
          <Highlight {...highlightProps}>
            <div className={classes.commentMeta}>
              <span className={classes.personName}>{creator.name}</span> {t('commented on')}{' '}
              <span className={classes.postTitle}>{postTitle}</span>
            </div>
          </Highlight>
          <span className={classes.date}>{timestamp}</span>
        </div>
        <CardImageAttachments attachments={attachments} linked className={classes.commentImages} />
        <CardFileAttachments attachments={attachments} className={classes.commentFiles} />
        <ClickCatcher groupSlug={slug}>
          <Highlight {...highlightProps}>
            <HyloHTML className={classes.commentBody} html={commentText} />
          </Highlight>
        </ClickCatcher>
        <div className={classes.commentFooter} />
      </div>
    </span>
  )
}
