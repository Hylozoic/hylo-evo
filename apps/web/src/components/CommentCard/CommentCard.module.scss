.link {
  width: 100%;
  &:hover {
    text-decoration: none;
  }
}

.comment-card {
  composes: bdy-lt-sm from 'css/typography.scss';
  $padding-left: $space-18x;
  border-radius: $space-1x;
  padding: $space-4x $space-5x;
  cursor: pointer;
  position: relative;

  &:hover {
    box-shadow: 0px 2px 6px 0px $color-black-10;
  }

  .comment-header {
    display: flex;
    align-items: center;
  }

  .comment-body {
    padding-left: 15px;
    margin-top: 10px;
  }

  .comment-images {
    margin-left: $padding-left;
    width: 50%;
    border-radius: 20px;
  }

  .comment-files {
    padding-left: $padding-left;
  }

  .comment-footer {
    display: flex;
    align-items: center;
    color: $color-caribbean-green;
    padding-left: $padding-left;

    a {
      color: $color-caribbean-green;

      &:hover {
        text-decoration: none;
      }
    }
  }

  &.expanded,
  &.expanded:hover {
    box-shadow: 0px 4px 6px 0px $color-black-10;
  }
}

.comment-body {
  composes: bdy-lt-sm from 'css/typography.scss';
}

.comment-meta {
  flex-grow: 3;
  margin-left: $space-3x;
}

.person-name {
  composes: person-name from 'css/typography.scss';
  font-size: 12px;
  line-height: 14px;

  &:hover {
    color: $color-rhino;
    text-decoration: none;
  }
}

.post-title {
  /* Might not be general enough for a global style */
  composes: person-name;
}

.date {
  color: $color-rhino-30;
  font-size: 12px;
  white-space: nowrap;
  position: absolute;
  top: 2px;
  right: 10px;
}

.reply-button {
  font-size: $space-3x;
  margin-right: $space-3x;
}

.profileImage {
  min-width: 2.75rem;
}


@media screen and (max-width: 425px) {
  .comment-card {
    padding: 10px;

    .comment-body {
      padding-left: 10px;
    }
  }
}
