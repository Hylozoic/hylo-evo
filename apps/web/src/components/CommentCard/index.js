import component from './CommentCard'
import connector from './CommentCard.connector'

export default connector(component)
