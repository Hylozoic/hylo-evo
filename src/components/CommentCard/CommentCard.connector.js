import { connect } from 'react-redux'
import { push } from 'react-router-redux'
import { postUrl } from 'util/navigation'

export function mapStateToProps (state, props) {
  return {
    routeParams: props.routeParams || {}
  }
}

export function mapDispatchToProps (dispatch, props) {
  return {
    showDetails: () =>
      dispatch(push(postUrl(props.comment.post.id, props.routeParams)))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)
