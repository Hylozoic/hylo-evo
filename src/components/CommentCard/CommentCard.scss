.comment-card {
  composes: bdy-lt-sm from 'css/typography.scss';
  $padding-left: $space-14x;

  border: 1px solid $color-cape-cod-10;
  border-radius: $space-1x;
  padding: $space-4x $space-5x;
  cursor: pointer;

  &:hover {
    box-shadow: 0px 2px 6px 0px $color-black-10;
  }

  .comment-header {
    display: flex;
    align-items: center;
  }

  .comment-body {
    padding-left: $padding-left;
  }

  .comment-footer {
    display: flex;
    align-items: center;
    color: $color-caribbean-green;
    padding-left: $padding-left;

    a {
      color: $color-caribbean-green;

      &:hover {
        text-decoration: none;
      }
    }
  }
}

.comment-body {
  composes: bdy-drk-lg from 'css/typography.scss';
}

.comment-meta {
  flex-grow: 3;
  margin-left: $space-3x;
}

.person-name {
  composes: person-name from 'css/typography.scss';
  &:hover {
    color: $color-rhino;
    text-decoration: none;
  }
}

.post-title {
  /* Might not be general enough for a global style */
  composes: person-name;
}

.date {
  color: $color-rhino-30;
}

.reply-button {
  font-size: $space-3x;
  margin-right: $space-3x;
}
