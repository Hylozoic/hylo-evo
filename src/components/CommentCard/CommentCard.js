import React from 'react'
import { Link } from 'react-router-dom'
import Icon from 'components/Icon'
import RoundImage from 'components/RoundImage'
import { postUrl } from 'util/index'
import { humanDate, present, sanitize } from 'hylo-utils/text'
import Highlight from 'components/Highlight'
import ClickCatcher from 'components/ClickCatcher'
import './CommentCard.scss'

export default function CommentCard ({ comment, shouldShowReply, expanded = true, highlightProps }) {
  const { creator, post, slug } = comment
  const postTitle = present(sanitize(post.title), { maxlength: 25, noP: true })
  const commentPresentOpts = {
    maxlength: expanded ? null : 144,
    noP: true,
    slug
  }
  const commentText = present(sanitize(comment.text), commentPresentOpts)

  return <Link to={postUrl(post.id, slug, {memberId: slug ? creator.id : null})} styleName='link'>
    <div styleName='comment-card'>
      <div styleName='comment-header'>
        <RoundImage url={creator.avatarUrl} large />
        <Highlight {...highlightProps}>
          <div styleName='comment-meta'>
            <span styleName='person-name'>{creator.name}</span> commented on&nbsp;
            <span styleName='post-title'>{postTitle}</span>
          </div>
        </Highlight>
        <span styleName='date'>{humanDate(comment.createdAt)}</span>
      </div>
      <ClickCatcher>
        <Highlight {...highlightProps}>
          <div styleName='comment-body' dangerouslySetInnerHTML={{__html: commentText}} />
        </Highlight>
      </ClickCatcher>
      <div styleName='comment-footer'>
        {shouldShowReply && <span><Icon styleName='reply-button' name='Reply' green /> Reply</span>}
      </div>
    </div>
  </Link>
}
