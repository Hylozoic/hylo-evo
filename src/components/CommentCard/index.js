import CommentCard from './CommentCard'
export default CommentCard
