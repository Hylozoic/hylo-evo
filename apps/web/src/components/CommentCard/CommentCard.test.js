import CommentCard from './CommentCard'
import { shallow } from 'enzyme'
import React from 'react'

const props = {
  comment: {
    text: '<p>text of the comment. a long one. text of the comment. a long one. text of the comment. a long one. text of the comment. a long one. text of the comment. a long one. text of the comment. a long one. Irrelevant Change</p>',
    creator: {
      id: 1,
      name: 'Joe Smith',
      avatarUrl: 'foo.jpg'
    },
    attachments: [],
    post: {
      id: 77,
      title: 'Awesome Sauce #hashtag'
    },
    createdAt: new Date()
  },
  expanded: false,
  highlightProps: { term: 'foo' }
}

it('matches last snapshot', () => {
  const wrapper = shallow(<CommentCard {...props} />)
  expect(wrapper).toMatchSnapshot()
})

it('matches last snapshot with different config', () => {
  const differentProps = {
    ...props,
    expanded: true
  }
  const wrapper = shallow(<CommentCard {...differentProps} />)
  expect(wrapper).toMatchSnapshot()
})

it('displays image an image attachments', () => {
  const comment = {
    ...props.comment,
    attachments: [
      { url: 'jam.png', attachmentType: 'image' }
    ]
  }
  const wrapper = shallow(<CommentCard {...props} comment={comment} />)
  expect(wrapper).toMatchSnapshot()
})
