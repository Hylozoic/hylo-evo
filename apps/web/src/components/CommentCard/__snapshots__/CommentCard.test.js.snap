// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays image an image attachments 1`] = `
<span
  data-stylename="link"
  onClick={[Function]}
>
  <div
    data-stylename="comment-card"
  >
    <div
      data-stylename="comment-header"
    >
      <RoundImage
        data-stylename="profileImage"
        url="foo.jpg"
      />
      <Highlight
        component="span"
        highlightClassName="highlight"
        term="foo"
      >
        <div
          data-stylename="comment-meta"
        >
          <span
            data-stylename="person-name"
          >
            Joe Smith
          </span>
           
          commented on
           
          <span
            data-stylename="post-title"
          >
            Awesome Sauce #hashtag
          </span>
        </div>
      </Highlight>
      <span
        data-stylename="date"
      >
        Commented just now
      </span>
    </div>
    <CardImageAttachments
      attachments={
        Array [
          Object {
            "attachmentType": "image",
            "url": "jam.png",
          },
        ]
      }
      data-stylename="comment-images"
      linked={true}
    />
    <CardFileAttachments
      attachments={
        Array [
          Object {
            "attachmentType": "image",
            "url": "jam.png",
          },
        ]
      }
      data-stylename="comment-files"
    />
    <ClickCatcher>
      <Highlight
        component="span"
        highlightClassName="highlight"
        term="foo"
      >
        <HyloHTML
          data-stylename="comment-body"
          html="<p>text of the comment. a long one. text of the comment. a long one. text of the comment. a long one. text of the comment. a long one. text of the …</p>"
        />
      </Highlight>
    </ClickCatcher>
    <div
      data-stylename="comment-footer"
    />
  </div>
</span>
`;

exports[`matches last snapshot 1`] = `
<span
  data-stylename="link"
  onClick={[Function]}
>
  <div
    data-stylename="comment-card"
  >
    <div
      data-stylename="comment-header"
    >
      <RoundImage
        data-stylename="profileImage"
        url="foo.jpg"
      />
      <Highlight
        component="span"
        highlightClassName="highlight"
        term="foo"
      >
        <div
          data-stylename="comment-meta"
        >
          <span
            data-stylename="person-name"
          >
            Joe Smith
          </span>
           
          commented on
           
          <span
            data-stylename="post-title"
          >
            Awesome Sauce #hashtag
          </span>
        </div>
      </Highlight>
      <span
        data-stylename="date"
      >
        Commented just now
      </span>
    </div>
    <CardImageAttachments
      attachments={Array []}
      data-stylename="comment-images"
      linked={true}
    />
    <CardFileAttachments
      attachments={Array []}
      data-stylename="comment-files"
    />
    <ClickCatcher>
      <Highlight
        component="span"
        highlightClassName="highlight"
        term="foo"
      >
        <HyloHTML
          data-stylename="comment-body"
          html="<p>text of the comment. a long one. text of the comment. a long one. text of the comment. a long one. text of the comment. a long one. text of the …</p>"
        />
      </Highlight>
    </ClickCatcher>
    <div
      data-stylename="comment-footer"
    />
  </div>
</span>
`;

exports[`matches last snapshot with different config 1`] = `
<span
  data-stylename="link"
  onClick={[Function]}
>
  <div
    data-stylename="comment-card expanded"
  >
    <div
      data-stylename="comment-header"
    >
      <RoundImage
        data-stylename="profileImage"
        url="foo.jpg"
      />
      <Highlight
        component="span"
        highlightClassName="highlight"
        term="foo"
      >
        <div
          data-stylename="comment-meta"
        >
          <span
            data-stylename="person-name"
          >
            Joe Smith
          </span>
           
          commented on
           
          <span
            data-stylename="post-title"
          >
            Awesome Sauce #hashtag
          </span>
        </div>
      </Highlight>
      <span
        data-stylename="date"
      >
        Commented just now
      </span>
    </div>
    <CardImageAttachments
      attachments={Array []}
      data-stylename="comment-images"
      linked={true}
    />
    <CardFileAttachments
      attachments={Array []}
      data-stylename="comment-files"
    />
    <ClickCatcher>
      <Highlight
        component="span"
        highlightClassName="highlight"
        term="foo"
      >
        <HyloHTML
          data-stylename="comment-body"
          html="<p>text of the comment. a long one. text of the comment. a long one. text of the comment. a long one. text of the comment. a long one. text of the comment. a long one. text of the comment. a long one. Irrelevant Change</p>"
        />
      </Highlight>
    </ClickCatcher>
    <div
      data-stylename="comment-footer"
    />
  </div>
</span>
`;
